protocol-units/settlement/mcr/setup/src/local/mod.rs

```diff
@@ -86,19 +91,23 @@ impl Local {
 		}
 
 		let anvil_addresses =
-				mcr_settlement_client::eth_client::read_anvil_json_file_addresses(
-					&*anvil_path,
-				).context("Failed to read Anvil addresses")?;
+			mcr_settlement_client::eth_client::read_anvil_json_file_addresses(&*anvil_path)
+				.context("Failed to read Anvil addresses")?;
 		if let Some(deploy) = &mut config.deploy {
-			deploy.mcr_deployment_account_private_key = anvil_addresses.get(0).ok_or(anyhow!("Failed to get Anvil address"))?.private_key.clone();
+			deploy.mcr_deployment_account_private_key = anvil_addresses
+				.get(0)
+				.ok_or(anyhow!("Failed to get Anvil address"))?
+				.private_key
+				.clone();
 		}
 		if let Some(testing) = &mut config.testing {
+			// Remove the old one if the config was existing.
+			testing.well_known_account_private_keys.clear();
```

Comment:
Wouldn't it be surprising for someone who pre-adds some keys to the config, to see them not present for the test run?

Reply:
In fact, I add the case when I've started several time process-compose without removing the .movement folder. If the config file already exists, without the clear, we add the new execution keys after the old one.

Reply:
This sounds messy to me. Should we at least have a flag so this behavior is explicitly enabled?

Reply:
Also, if we have a test that reuses the setup, this should be a special case API in setup that does not get entangled with the normal case of running once on a clean `.movement`?

Reply:
I think I'm just confused by this code. `config` is passed to this function, presumably read from some configuration on file in the test setup. The configuration is further modified by setup, then returned, to be used as the effective configuration by the test. That returned config should never be passed again to `Local::setup`, instead the original config source should be used.

Reply:
OK, this is arguing about minor things and practically, the list of private keys should always be in sync with `anvil.json`, which this achieves.